src/zenml/integrations/seldon/model_deployers/seldon_model_deployer.py

```diff
@@ -362,3 +367,54 @@ def delete_model_server(
         if len(services) == 0:
             return
         services[0].stop(timeout=timeout, force=force)
+
+    def prepare_custom_deployment_image(
+        self,
+        stack: "Stack",
+        pipeline_name: str,
+        step_name: str,
+        requirements: List[str],
+        model_name: str,
+        entrypoint_command: List[str],
+    ) -> str:
+        """Prepare a custom Docker image for a Seldon Core model deployment.
+
+        This method prepares a custom Docker image for a Seldon Core model
+        deployment. The image is built and pushed to the container registry
+        configured for the active stack. The image digest is returned.
+
+        Args:
+            stack: the stack that the model deployment is part of.
+            pipeline_name: name of the pipeline that the deployed model was part
+                of.
+            step_name: name of the pipeline model deployment step that deployed
+                the model.
+            model_name: name of the deployed model.
+            requirements: list of requirements to be included in the Dockerfile.
+            entrypoint_command: command to be executed by the Docker image.
+
+        Returns:
+            The image digest of the prepared Docker image.
+        """
+
+        container_registry = stack.container_registry
+
+        if not container_registry:
+            raise RuntimeError("Missing container registry")
+
+        registry_uri = container_registry.uri.rstrip("/")
+        image_name = f"{registry_uri}/zenml-seldon-custom-deploy:{pipeline_name}-{step_name}"
```

Comment:
Might make sense to track this main stub of the name 'zenml-seldon-custom-deploy' in a constnat somewhere, in case we want to change this at some point in the future